aten/src/ATen/native/cuda/jit_utils.cpp

```diff
@@ -951,48 +995,51 @@ std::string generate_code(
       desc.nOutputs,
       desc.f,
       desc.name,
-      typeName(desc.f_inputs_type),
-      typeName(toOpMathType(desc.f_inputs_type)),
-      typeName(desc.result_type),
+      desc.f_inputs_type,
+      toOpMathType(desc.f_inputs_type),
+      desc.result_type,
       contiguous,
       dynamic_casting,
       scalar_pos,
       extra_args_typenames,
+      thread_work_size,
       vectorized,
       vec_size,
       return_by_ref);
 }
 
-//FIXME - this are defined in Loops.cuh, but including Loops.cuh here would lead to circular includes Loops.cuh -> CUDALoops.cuh -> jit_utils.h -> Loops.cuh
-#define THREAD_WORK_SIZE 4
-constexpr int thread_work_size = THREAD_WORK_SIZE;
-
 std::string generate_code(
     int nInputs,
     int nOutputs,
     const std::string& func_,
     const std::string& name,
-    const std::string& f_inputs_type,
-    const std::string& compute_type,
-    const std::string& result_type,
+    const c10::ScalarType& f_inputs_type,
+    const c10::ScalarType& compute_type,
+    const c10::ScalarType& result_type,
```

Comment:
Why did these change from std::string to c10::ScalarType?  It forces a lot of other changes where you now create new variables named with *_str suffix that you then use in the rest of the file.  Was that necessary?  What was gained?

Reply:
Yes, there are not required now. 
This was needed to compute "thread_work_size". It is now a parameter that will be passed by caller.
Will make a new revision with the fix